pkg/jobs/jobsprofiler/profilerconstants/constants.go

```diff
@@ -10,6 +10,32 @@
 
 package profilerconstants
 
-// DSPDiagramInfoKeyPrefix is the prefix of the info key used for rows that
-// store the DistSQL plan diagram for a job.
-const DSPDiagramInfoKeyPrefix = "dsp-diag-url-"
+import (
+	"fmt"
+
+	"github.com/cockroachdb/errors"
+)
+
+const DSPDiagramInfoKeyPrefix = "~dsp-diag-url-"
+
+// MakeDSPDiagramInfoKey constructs an ephemeral DSP diagram info key.
+func MakeDSPDiagramInfoKey(parts ...any) (string, error) {
+	if len(parts) != 1 {
+		return "", errors.AssertionFailedf("expected 1 part got %d", len(parts))
+	}
+	return fmt.Sprintf("%s%d", DSPDiagramInfoKeyPrefix, parts[0]), nil
+}
+
+// NodeProcessorProgressInfoKeyPrefix is the prefix of the info key used for
+// rows that store the per node, per processor progress for a job.
+const NodeProcessorProgressInfoKeyPrefix = "~node-processor-progress-"
+
+// MakeNodeProcessorProgressInfoKey returns the info_key used for rows that
+// store the per node, per processor progress for a job.
+func MakeNodeProcessorProgressInfoKey(parts ...any) (string, error) {
```

Comment:
Why …any? could just pass two strings and an int?

Reply:
doh yeah copied it over from when this was interface, will fix this up.